expression/core/union.py

```diff
@@ -125,7 +126,7 @@ def __init__(self, tag: Tag[Any], __value: Any = None, **kwargs: Any) -> None:
 
         self.name = getattr(self.__class__, "_tags")[tag.tag]
 
-    def __match__(self, pattern: _T) -> Iterable[_T]:
+    def __match__(self, pattern: Union[_T, Type[_T]]) -> Iterable[_T]:
```

Comment:
Q: what is the difference between a _T and and Type[_T] here?

Reply:
`_T` would represent that you give a value like 42 or "test" and an argument. `Type[_T]` would represent that you give the type `int` or `str` as an argument (not the value). This is because we want to be able to pattern match on a type as well as a given value.

Reply:
Aha, my hunch was correct then, thanks for clarifying!